### PR TITLE
Add adaptive learning rate and adam

### PR DESCRIPTION
**Pull request by moduIo**
Adds:
1. Adam
2. Adaptive STT learning rate

### CODE REVIEW

File: tasks/train.py [hunk 1]
```diff
@@ -73,6 +76,7 @@ def train(args):
 
             loss.backward()
             opt.step()
+            scheduler.step()
```
Comment (HershGupta25) [inline tasks/train.py:79]: Do we need to implement the gradient clipping? Like below from the paper

Before updating the parameters, gradients are clipped with
respect to the global gradient norm (i.e., clip by global norm)
with a maximum norm value of 1.0

Comment (moduIo) [reply tasks/train.py]: Good callout, I added gradient clipping to 1
